pkg/storage/cedar_system_bookmark.go

```diff
@@ -102,3 +80,38 @@ func (s *Store) DeleteCedarSystemBookmark(ctx context.Context, cedarSystemBookma
 
 	return cedarSystemBookmark, nil
 }
+
+func (s *Store) FetchCedarSystemIsBookmarkedByCedarSystemIDs(ctx context.Context, bookmarkRequests []models.BookmarkRequest) ([]bool, error) {
+	// build lists for multiple `where` clauses
+	var (
+		euaUserIDs = make([]string, len(bookmarkRequests))
+		systemIDs  = make([]string, len(bookmarkRequests))
+	)
+
+	for i, req := range bookmarkRequests {
+		euaUserIDs[i] = req.EuaUserID
+		systemIDs[i] = req.CedarSystemID
+	}
+
+	var results []models.BookmarkRequest
+	if err := selectNamed(ctx, s, &results, sqlqueries.CedarBookmarkSystemsForm.SelectByCedarSystemIDs, args{
+		"eua_user_ids":     pq.Array(euaUserIDs),
+		"cedar_system_ids": pq.Array(systemIDs),
+	}); err != nil {
+		return nil, err
+	}
+
+	store := map[models.BookmarkRequest]struct{}{}
+
+	for _, result := range results {
+		store[result] = helpers.EmptyStruct
+	}
+
+	var out []bool
```

Comment:
```suggestion
	// Order the results map by the input keys (`bookmarkRequests`)
	var out []bool
```

I think comments like this can be helpful to just describe little blocks like this

Reply:
thanks!